### ISSUE TITLE
Console error when trying to retrieve posts from other websites

### ISSUE THREAD

**Issue by saxamaphone69** [TuxedoTako/4chan-xt#23]
Playing around with things and I managed to get this error pop up in the console:

```
userscript.html?name=4chan-XT.user.js&id=10d5d7a8-76a3-49ab-a7bd-3d6108ba965d:13504
Uncaught RangeError: Maximum call stack size exceeded
    at Object.parseJSON (userscript.html?name=4chan-XT.user.js&id=10d5d7a8-76a3-49ab-a7bd-3d6108ba965d:13504:24)
    at Object.parseJSON (userscript.html?name=4chan-XT.user.js&id=10d5d7a8-76a3-49ab-a7bd-3d6108ba965d:13504:24)
    at Object.parseJSON (userscript.html?name=4chan-XT.user.js&id=10d5d7a8-76a3-49ab-a7bd-3d6108ba965d:13504:24)
    at Object.parseJSON (userscript.html?name=4chan-XT.user.js&id=10d5d7a8-76a3-49ab-a7bd-3d6108ba965d:13504:24)
    at Object.parseJSON (userscript.html?name=4chan-XT.user.js&id=10d5d7a8-76a3-49ab-a7bd-3d6108ba965d:13504:24)
    at Object.parseJSON (userscript.html?name=4chan-XT.user.js&id=10d5d7a8-76a3-49ab-a7bd-3d6108ba965d:13504:24)
    at Object.parseJSON (userscript.html?name=4chan-XT.user.js&id=10d5d7a8-76a3-49ab-a7bd-3d6108ba965d:13504:24)
    at Object.parseJSON (userscript.html?name=4chan-XT.user.js&id=10d5d7a8-76a3-49ab-a7bd-3d6108ba965d:13504:24)
    at Object.parseJSON (userscript.html?name=4chan-XT.user.js&id=10d5d7a8-76a3-49ab-a7bd-3d6108ba965d:13504:24)
    at Object.parseJSON (userscript.html?name=4chan-XT.user.js&id=10d5d7a8-76a3-49ab-a7bd-3d6108ba965d:13504:24)
```

which takes you here:

```js
Build: {
      parseJSON(data, board) {
        const o = this.parseJSON(data, board); // <---
        if (data.ext === 'deleted') {
          delete o.file;
          $$1.extend(o, {
            files: [],
            fileDeleted: true,
            filesDeleted: [0]
          });
        }
        if (data.extra_files) {
          let file;
          for (let i = 0; i < data.extra_files.length; i++) {
            var extra_file = data.extra_files[i];
            if (extra_file.ext === 'deleted') {
              o.filesDeleted.push(i);
            } else {
              file = this.parseJSONFile(data, board);
              o.files.push(file);
            }
          }
          if (o.files.length) {
            o.file = o.files[0];
          }
        }
        return o;
      },

      parseComment(html) {
        html = html
          .replace(/<br\b[^<]*>/gi, '\n')
          .replace(/<[^>]*>/g, '');
        return $$1.unescape(html);
      }
    },
```

Here's the workflow to get it to pop up for me. Chrome and Tampermonkey.
1. Add [this](https://sushigirl.us/kitchen/res/755.html) to your watchlist (I'd assume any of the other "supported" sites applies the same)
2. Head back over to 4chan and attempt to refresh the thread watcher
3. Tampermonkey warns you about the request to a different website. I click 'Allow Once'
![image](https://github.com/TuxedoTako/4chan-xt/assets/877480/3e924df7-3d59-4cd4-81b3-8f2259ed7331)
4. You get the error above. Doesn't seem to actually impose/prevent anything, though I do notice it doesn't display a `(0)` in the title of the thread in the Thread Watcher.

From what I can tell, no such error on normal 4chan X; so potentially something that happened during the TS conversion?

**Comment by saxamaphone69** [comment]
Speaking of other websites. Probably a separate issue.

```
"Thread Stats" crashed on node Thread No.1028 (arcade). [[report](https://github.com/TuxedoTako/4chan-xt/issues)]
TypeError: Cannot set properties of null (setting 'textContent')
(4chan XT TuxedoTako v2.4.1 userscript on blink at 25837,25793,372,27169,26993,13383,26956,21535)
```

```
userscript.html?name=4chan-XT.user.js&id=10d5d7a8-76a3-49ab-a7bd-3d6108ba965d:27247 "Thread Stats" crashed on node Thread No.1028 (arcade). TypeError: Cannot set properties of null (setting 'textContent')
    at Object.update (chrome-extension://dhdgffkkebhmkfjojejmpbldmpobfkfo/userscript.html?name=4chan-XT.user.js&id=10d5d7a8-76a3-49ab-a7bd-3d6108ba965d:25837:30)
    at Thread.node (chrome-extension://dhdgffkkebhmkfjojejmpbldmpobfkfo/userscript.html?name=4chan-XT.user.js&id=10d5d7a8-76a3-49ab-a7bd-3d6108ba965d:25793:19)
    at Callbacks.execute (chrome-extension://dhdgffkkebhmkfjojejmpbldmpobfkfo/userscript.html?name=4chan-XT.user.js&id=10d5d7a8-76a3-49ab-a7bd-3d6108ba965d:372:23)
    at Object.callbackNodes (chrome-extension://dhdgffkkebhmkfjojejmpbldmpobfkfo/userscript.html?name=4chan-XT.user.js&id=10d5d7a8-76a3-49ab-a7bd-3d6108ba965d:27169:12)
    at initThread (chrome-extension://dhdgffkkebhmkfjojejmpbldmpobfkfo/userscript.html?name=4chan-XT.user.js&id=10d5d7a8-76a3-49ab-a7bd-3d6108ba965d:26993:14)
    at Object.awaitBoard (chrome-extension://dhdgffkkebhmkfjojejmpbldmpobfkfo/userscript.html?name=4chan-XT.user.js&id=10d5d7a8-76a3-49ab-a7bd-3d6108ba965d:13383:16)
    at initReady (chrome-extension://dhdgffkkebhmkfjojejmpbldmpobfkfo/userscript.html?name=4chan-XT.user.js&id=10d5d7a8-76a3-49ab-a7bd-3d6108ba965d:26956:25)
    at execTask (chrome-extension://dhdgffkkebhmkfjojejmpbldmpobfkfo/userscript.html?name=4chan-XT.user.js&id=10d5d7a8-76a3-49ab-a7bd-3d6108ba965d:21535:7)
parseError @ userscript.html?name=4chan-XT.user.js&id=10d5d7a8-76a3-49ab-a7bd-3d6108ba965d:27247
15:55:32.633 
```

Not that I go to these other sites, but noticed it when loading into a thread.

**Comment by TuxedoTako** [comment]
Thanks for testing XT with websites outside of 4chan more than I did. Both should be fixed in [2.4.2](https://github.com/TuxedoTako/4chan-xt/releases). Can you check?

**Comment by saxamaphone69** [comment]
That seems to have done it 👍
